retrieval/aggregator.go

```diff
@@ -0,0 +1,158 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package retrieval
+
+import (
+	"context"
+	"math"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	promlabels "github.com/prometheus/prometheus/pkg/labels"
+	"github.com/prometheus/tsdb/labels"
+	"go.opencensus.io/stats"
+	"go.opencensus.io/stats/view"
+)
+
+var statsRecord = stats.Record
+
+const metricPrefix = "prometheus_sidecar/aggregated_counters/"
+
+// CounterAggregator provides the 'aggregated counters' feature of the sidecar.
+// It can be used to export a sum of multiple counters from Prometheus to
+// Stackdriver as a single cumulative metric.
```

Comment:
Nic, I'd say "as a single counter metric", and perhaps add "with delta encoding", because Stackdriver makes a distinction between cumulative and delta metrics, and this code is changing cumulative into delta.

Reply:
I don't think this is true. These metrics are reported by OpenCensus to Stackdriver as regular cumulative metrics, with 'start time' remaining constant while the process is running.

Reply:
Thanks for the explanation, now I have a better understanding.